Add several tests

Semigradsky opened the pull request.

Semigradsky commented:
It looks like the `shallowEqual` property from test suites is not used.

planttheidea commented:
I can kind of glean the goal of this PR from the changes, but just to confirm ... certain specific failures in other libraries were not tested here, and you are adding coverage for those tests? It would be very helpful if a description were added to give context for the change.

> It looks like the `shallowEqual` property from test suites is not used.

It is a little opaque, but it is used when [iterating over the entire suite and comparing](https://github.com/planttheidea/fast-equals/blob/master/__tests__/index.ts#L83). The renaming doesn't help there.

Semigradsky commented:
Yeah, `Object.create(null)` is a pretty old feature (modern equivalent to `{ __proto__: null }`) and I was surprised that some libraries can't handle this. I added a test to avoid issues here.

About sparse arrays - I think it is not a very commonly used feature, I just added a test to represent how it works.
